misc/controller/misc/log_weather_wu.sh

```diff
@@ -0,0 +1,173 @@
+CURL=/usr/bin/curl
+NC=/bin/nc
+
+weatherfile="/tmp/wunderground.$PWS"
+## Get data from OpenWeatherMap
+$CURL --data "" "$WU" > "$weatherfile"
+test -s "$weatherfile" || exit 1 ## file is empty
+## Grep the weather data from answare
+## $ echo "foo start blah blah blah stop bar" | sed 's/.*start \(.*\) stop.*/\1/'
+## result: blah blah blah
+last_update="`grep 'observation_epoch":"' "$weatherfile"|sed -e 's/.*":"\(.*\)".*/\1/'`"
+temperature="`grep '"temp_c":' "$weatherfile"|sed -e 's/.*c":\(.*\),.*/\1/'`"
+pressure="`grep '"pressure_mb":"' "$weatherfile"|sed -e 's/.*mb":"\(.*\)".*/\1/'`"
+relative_humidity="`grep '"relative_humidity":"' "$weatherfile"|sed -e 's/.*":"\(.*\)%.*/\1/'`"
+dewpoint="`grep '"dewpoint_c":' "$weatherfile"|sed -e 's/.*c":\(.*\),.*/\1/'`"
+winddirection="`grep '"wind_degrees":' "$weatherfile"|sed -e 's/.*s":\(.*\),.*/\1/'`"
+wind="`grep '"wind_kph":' "$weatherfile"|sed -e 's/.*kph":\(.*\),.*/\1/'`"
+solar="`grep '"solarradiation":"' "$weatherfile"|sed -e 's/.*on":"\(.*\)".*/\1/'`"
+uvindex="`grep '"UV":"' "$weatherfile"|sed -e 's/.*UV":"\(.*\)",".*/\1/'`"
+
+## Convert the actual computer time in a timestamp
+dateA=`date +%Y-%m-%d`
+timeA=`date +%H:%M:%S`
+## Timestamp actual computer time
+## timestamp="`date -d "$timeA" "+%s000"`"
+## Timestamp from OpenWeatherMap
+timestamp=$last_update"000"
+
+### Debug
+ echo -e "LocalTime: "`date`
+ echo -e "UTC Time: "`date -u`
+ echo "PWS: "$PWS
+ echo "Temp: "$temperature"°C"
+ echo "Humi: "$relative_humidity"%"
+ echo "Pres: "$pressure"hPa"
+ echo "Dewp: "$dewpoint"°C"
+ echo "WDir: "$winddirection"°"
+ echo "Wind: "$wind"km/h"
+ echo "Solar: "$solar"W/m2"
+ echo "UV: "$uvindex""
+ echo "Date: "$dateA $timeA
+ echo "Last Update: "$last_update
+ echo "Timestamp: "$timestamp
+### Debug
+
+
+# Save the data with or withou timestamp from WeatherUnderground
+# when exist a UUID
+if [ -n "$APPID" ]; then
+  if [ $TS = "0" ]; then
+  echo "Saving all values with local PC time"
+  else
+  echo "Saving all values with update time from WeatherUnderground"
+  fi
+  if [ -n "$UUID_temperature" ]; then
+  echo "Save Temp with UUID: "$UUID_temperature" in database"
+    if [ $TS = "0" ]; then
+    $CURL --data "" "$URL/data/$UUID_temperature.json?value=$temperature" &>/dev/null
+    else
+    $CURL --data "" "$URL/data/$UUID_temperature.json?ts=$timestamp&value=$temperature" &>/dev/null
+    fi
+  fi
+  if [ -n "$UUID_pressure" ]; then
+  echo "Save Pressure with UUID: "$UUID_pressure" in database"
+    if [ $TS = "0" ]; then
+    $CURL --data "" "$URL/data/$UUID_temperature.json?value=$pressure" &>/dev/null
+    else
+    $CURL --data "" "$URL/data/$UUID_pressure.json?ts=$timestamp&value=$pressure" &>/dev/null
+    fi
+  fi
+  if [ -n "$UUID_relative_humidity" ]; then
+  echo "Save Humidity  with UUID: "$UUID_relative_humidity" in database"
+    if [ $TS = "0" ]; then
+    $CURL --data "" "$URL/data/$UUID_relative_humidity.json?value=$relative_humidity" &>/dev/null

```

Comment:
--data
(HTTP) Sends the specified data in a POST request to the HTTP server......
-d, --data is the same as --data-ascii. --data-raw is almost the same......
War so auch im alten Script. Hab nicht drüber nachgedacht, weil es funktionierte. Der String wird als ASCII über POST gesendet. Wie würdest Du es denn machen und warum?